[CUBRIDMAN-131] Add manual for REGEXP related functions using RE2 library

hgryoo opened the pull request:
http://jira.cubrid.org/browse/CUBRIDMAN-131

- Add a description of the system parameter, 'regex_engine'
- Add a description of compatibility consideration from CUBRID 11.2
- Add C++ <regex> library caveats warning section
- Revise the Special Pattern Characters table
- Add the Unicode Character Class section

hgryoo committed:
fix some error and typo in english manual

hgryoo committed:
add korean manual draft

hgryoo committed:
fix typo

hgryoo committed:
finish korean translation, ready to review

hgryoo committed:
regexp_engine system paramter, c++ regex library caveats

hgryoo committed:
Merge branch 'develop' into CUBRIDMAN-131

hgryoo committed:
add description related to RE2 compatibility, especially unicode class

hgryoo committed:
Merge branch 'CUBRIDMAN-131' of https://github.com/hgryoo/cubrid-manual into CUBRIDMAN-131

youngjinj commented on ko/sql/function/regex_fn.rst:
```diff
@@ -12,20 +12,23 @@
-ECMAScript 정규 표현식 문법
-==============================================
+정규 표현식 문법
+=======
```
I hope to resolve the warning.
```suggestion
================
```

hgryoo replied:
Thank you!!

hgryoo committed:
Update ko/sql/function/regex_fn.rst

Co-authored-by: Joo Youngjin <68979537+youngjinj@users.noreply.github.com>

hgryoo committed:
remove redundant spaces

hgryoo committed:
Merge branch 'CUBRIDMAN-131' of https://github.com/hgryoo/cubrid-manual into CUBRIDMAN-131

youngjinj commented:
I hope to resolve the warning.
```
$ git diff
diff --git a/ko/sql/function/regex_fn.rst b/ko/sql/function/regex_fn.rst
index 4515ac9..ba611bf 100644
--- a/ko/sql/function/regex_fn.rst
+++ b/ko/sql/function/regex_fn.rst
@@ -15,7 +15,7 @@
 .. _regex-syntax:

 정규 표현식 문법
-=======
+================

 정규 표현식을 구현하기 위해 CUBRID는 Google RE2 라이브러리와 C++ <regex> 표준 라이브러리를 사용한다.
 **regexp_engine** 시스템 파라미터를 설정하여 정규 표현식 라이브러리를 선택할 수 있고 Google RE2 라이브러리를 기본으로 사용한다.
@@ -75,18 +75,18 @@
 +----------------+----------------------------------------------------------------------------------------------------------+
 | \\r            | 캐리지 리턴 문자 (\\u000D 와 동일)                                                                       |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \\c\ *letter*  | 코드 단위 값이 *letter*의 코드 단위 값을                                                                 |
+| \\c\ *letter*  | 코드 단위 값이 *letter*\의 코드 단위 값을                                                                |
 |                | 32로 나눈 나머지와 같은 제어 코드 문자.                                                                  |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \\x\ *hh*      | 코드 단위 값이 2개의 16진수 *hh*에 해당하는 16진수 값을 갖는 문자                                        |
+| \\x\ *hh*      | 코드 단위 값이 2개의 16진수 *hh*\에 해당하는 16진수 값을 갖는 문자                                       |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \\u\ *hhhh*    | 코드 단위 값이 4자리 16진수 *hhhh*에 해당하는 16진수 값을 갖는 문자 *C++ <regex>만 지원*                 |
+| \\u\ *hhhh*    | 코드 단위 값이 4자리 16진수 *hhhh*\에 해당하는 16진수 값을 갖는 문자 *C++ <regex>만 지원*                |
 +----------------+----------------------------------------------------------------------------------------------------------+
 | \\x\ {*digits*}| *digits* 값에 해당하는 16진수 값을 갖는 문자 *Google RE2만 지원*                                         |
 +----------------+----------------------------------------------------------------------------------------------------------+
 | \\0            | 널 문자 (\\u0000와 동일). *C++ <regex>만 지원*                                                           |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \\\ *num*      | 여는 괄호가 *num*번째인 부분 일치의 결과. 자세한 내용은 아래 그룹을 참조.                                |
+| \\\ *num*      | 여는 괄호가 *num*\번째인 부분 일치의 결과. 자세한 내용은 아래 그룹을 참조.                               |
 +----------------+----------------------------------------------------------------------------------------------------------+
 | \\d            | 10진수 문자 ([[:digit:]] 와 동일).                                                                       |
 +----------------+----------------------------------------------------------------------------------------------------------+
@@ -104,11 +104,11 @@
 | \\\ *character*| | 위의 특수 문자 시퀀스를 형성하는 문자를 제외한 모든 문자를 이스케이프 한다.                            |
 |                | | 필요한 대상: ^ $ \\ . * + ? ( ) [ ] { } \|                                                             |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \[\ *class*\]  | *class*의 일부인 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                              |
+| \[\ *class*\]  | *class*\의 일부인 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                             |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \[^\ *class*\] | *class*의 일부가 아닌 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                         |
+| \[^\ *class*\] | *class*\의 일부가 아닌 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                        |
 +----------------+----------------------------------------------------------------------------------------------------------+
-| \\p{*class*}   | *class*의 유니코드 범위를 가지는 문자열. 아래의 :ref:`regex-unicode-character-class`\를 참조.            |
+| \\p{*class*}   | *class*\의 유니코드 범위를 가지는 문자열. 아래의 :ref:`regex-unicode-character-class`\를 참조.           |
 +----------------+----------------------------------------------------------------------------------------------------------+

 .. code-block:: sql
@@ -164,11 +164,11 @@
 +----------------+-------------------------------------------------------------------------+
 | ?              | 선행하는 문자 또는 패턴이 선택사항 (0회 또는 1회 일치)                  |
 +----------------+-------------------------------------------------------------------------+
-| {*num*}        | 선행하는 문자 또는 패턴이 정확히 *num* 회 일치                          |
+| {*num*}        | 선행하는 문자 또는 패턴이 정확히 *num*\회 일치                          |
 +----------------+-------------------------------------------------------------------------+
-| {*num,*}       | 선행하는 문자 또는 패턴이 정확히 *num* 회 또는 그 이상 일치             |
+| {*num,*}       | 선행하는 문자 또는 패턴이 정확히 *num*\회 또는 그 이상 일치             |
 +----------------+-------------------------------------------------------------------------+
-| {*min,max*}    | 선행하는 문자 또는 패턴이 최소 *min* 회 일치하지만 *max*를 넘지 않음    |
+| {*min,max*}    | 선행하는 문자 또는 패턴이 최소 *min*\회 일치하지만 *max*\를 넘지 않음   |
 +----------------+-------------------------------------------------------------------------+

 .. code-block:: sql
@@ -225,7 +225,7 @@
       1

 기본적으로 이러한 모든 수량자는 가능한 한 조건을 충족하는 많은 문자를 사용하는 *greedy* 방식으로 수행된다.
-그리고 이 동작은 한정자 뒤에 물음표(?)를 추가하여 *non-greedy*로 재정의할 수 있다.
+그리고 이 동작은 한정자 뒤에 물음표(?)를 추가하여 *non-greedy*\로 재정의할 수 있다.

 .. code-block:: sql

@@ -371,7 +371,7 @@
     ==============================
       0

-정규식은 구분자 연산자(alternatvies, |)로 간단히 구분하여 여러 대체 패턴을 포함할 수 있다.
+정규식은 구분자 연산자(alternatvies, \|)로 간단히 구분하여 여러 대체 패턴을 포함할 수 있다.
 정규식은 대체 패턴이 일치하는 경우 일치하는 즉시 매칭한다.
 하위 패턴(그룹 또는 어서션)은 구분자 연산자를 사용하여 여러 다른 패턴을 구분할 수도 있다.

@@ -538,7 +538,7 @@ POSIX 기반 문자 클래스(*[:classname:]*)는 아래와 같이 문자의 범
 ~~~~~~~~~~~~~~~~~~~~~~~

 유니코드 문자 클래스(*\\p{classname}*)는 Google RE2 에서만 제공한다. 아래의 표와 같이 클래스 이름을 지정할 수 있다.
-예를 들어 한글 문자를 매칭하기 위해서는 **\\p{Hangul}**을 사용한다.
+예를 들어 한글 문자를 매칭하기 위해서는 **\\p{Hangul}**\을 사용한다.

 +-------------------------+
 | 유니코드 문자 클래스    |
@@ -886,8 +886,8 @@ POSIX 기반 문자 클래스(*[:classname:]*)는 아래와 같이 문자의 범
 REGEXP, RLIKE
 =============

-**REGEXP**, **RLIKE**\ 는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*이 *pattern*과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
-*expression* 또는 *pattern*이 **NULL**이면 **NULL**을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**을 사용하는 세 번째 구문과 동일한 의미를 갖는다.
+**REGEXP**, **RLIKE**\는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*\이 *pattern*\과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
+*expression* 또는 *pattern*\이 **NULL**\이면 **NULL**\을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**\을 사용하는 세 번째 구문과 동일한 의미를 갖는다.

 ::

@@ -944,7 +944,7 @@ REGEXP_COUNT

 .. function:: REGEXP_COUNT (string, pattern_string [, position [, match_type]])

-    **REGEXP_COUNT** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*의 매칭 횟수를 반환한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
+    **REGEXP_COUNT** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\의 매칭 횟수를 반환한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.

     :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
     :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
@@ -1095,7 +1095,7 @@ REGEXP_LIKE

 .. function:: REGEXP_LIKE (string, pattern_string [, match_type])

-    **REGEXP_LIKE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*을 검색한다. 정규식 패턴이 *pattern_string*의 어느 곳에서나 일치하면 1이 반환된다. 그렇지 않으면 0이 반환된다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
+    **REGEXP_LIKE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\을 검색한다. 정규식 패턴이 *pattern_string*\의 어느 곳에서나 일치하면 1이 반환된다. 그렇지 않으면 0이 반환된다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.

     :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
     :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
@@ -1156,7 +1156,7 @@ REGEXP_REPLACE

 .. function:: REGEXP_REPLACE (string, pattern_string, replacement_string [, position [, occurrence [, match_type]]])

-    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*을 검색하여 문자열 *replacement_string*으로 바꾼다. **NULL**이 인수로 지정된 경우 **NULL**을 반환한다.
+    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\을 검색하여 문자열 *replacement_string*\으로 바꾼다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.

     :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
     :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
@@ -1235,7 +1235,7 @@ REGEXP_SUBSTR

 .. function:: REGEXP_SUBSTR (string, pattern_string [, position [, occurrence [, match_type]]])

-    **REGEXP_SUBSTR** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*에 일치하는 문자열을 추출한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
+    **REGEXP_SUBSTR** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\에 일치하는 문자열을 추출한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.

     :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
     :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
```

youngjinj commented on en/sql/function/regex_fn.rst:
```diff
@@ -381,7 +396,8 @@ Character classes
 
-**Individual characters** 
+Individual characters
+~~~~~~~~~~~~~~~~~~~~~~~
```
I don't know the cause, but the number of "~" must be reduced to create a hyperlink.
```suggestion
~~~~~~~~~~~~~~~~~~~~~
```

youngjinj commented on en/sql/function/regex_fn.rst:
```diff
@@ -37,6 +40,14 @@ The following sub-sections describes supported regular expression grammars with
   
```
It's trivial.
I think it should have a period dot.
```
\3. The Spencer library matches line-terminator characters for the dot operator (.) But it does not.

operator (.) → operator (.).
```

youngjinj commented on en/sql/function/regex_fn.rst:
```diff
@@ -442,7 +459,10 @@ For example, "[a-z]" matches any alphabet letter whereas "[0-9]" matches any sin
+.. _regex-posix-character-class:
+
+POSIX-based character classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
In en/sql/function/regex_fn.rst there is no following from line 502.
```
[:d:], [:w:] 및 [:s:]는 ECMAScript 문법의 확장이다. C++ <regex> 에서만 사용할 수 있다.

+------------+-----------------------------------------+
| 클래스     | 설명                                    |
+============+=========================================+
| [:d:]      | 숫자 (0-9)                              |
+------------+-----------------------------------------+
| [:w:]      | 단어                                    |
+------------+-----------------------------------------+
| [:s:]      | 공백                                    |
+------------+-----------------------------------------+
```

youngjinj commented on ko/sql/function/regex_fn.rst:
```diff
@@ -541,6 +921,7 @@ REGEXP, RLIKE
     -- SELECT 리스트에서 REGEXP를 사용하는 경우 괄호로 묶어야 한다.
     -- 단, WHERE 절에서는 괄호가 필요하지 않다.
     -- BINARY와 함께 사용하는 경우를 제외하고 대소문자를 구분하지 않는다.
+
```
It's trivial.
Only the ko version has blank lines.

youngjinj replied:
It's trivial.
Only the ko version has blank lines.
```suggestion
```

youngjinj commented:
Sorry for not reviewing properly before.

The example below seems to need fixing.

1. en/sql/function/regex_fn.rst:1267
   **AS-IS**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_SUBSTR('삼성로, 강남구, 서울특별시','[[:alpha:]]+',1,2);
       
   ::
   
       regexp_substr('삼성로, 강남구, 서울특별시', [[:alpha:]]+', 1, 2);
       ======================
         '강남구'
   ```
   **TO-BE**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_SUBSTR('Samseong-ro, Gangnam-gu, Seoul','[[:alpha:]]+',1,2);
       
   ::
   
       regexp_substr('Samseong-ro, Gangnam-gu, Seoul', [[:alpha:]]+', 1, 2);
       ======================
         'Gangnam-gu'
   ```
2. ko/sql/function/regex_fn.rst:1267
   **AS-IS**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_SUBSTR('삼성로, 강남구, 서울특별시','[[:alpha:]]+',1,2);
       
   ::
   
       regexp_substr('삼성로, 강남구, 서울특별시', [[:alpha:]]+', 1, 2);
       ======================
         '강남구'
   ```
   **TO-BE**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_SUBSTR('삼성로, 강남구, 서울특별시','\p{Hangul}+',1,2);
       
   ::
   
       regexp_substr('삼성로, 강남구, 서울특별시', '\p{Hangul}+', 1, 2);
       ======================
         '강남구'
   ```

youngjinj commented:
Sorry for not reviewing properly before.

The examples below do not work as expected.
Below is for "en". "ko" needs to be checked.

1. Example of "Special Pattern Characters" (140 lines)
   **AS-IS**
   ```
   .. code-block:: sql
   
       -- \n : match a special character, when no_backslash_escapes=no
       SELECT ('new\nline' REGEXP 'new
       line');
   
   ::
       
       ('new
       line' regexp 'new
       line')
       =====================================
         1
   ```
   **TO-BE**
      - Use new line character
   ```
   .. code-block:: sql
   
       -- \n : match a special character, when no_backslash_escapes=no
       SELECT ('new
       line' REGEXP 'new
       line');
   
   ::
       
       ('new
       line' regexp 'new
       line')
       =====================================
         1
   ```

2. Example of "Groups" (255 lines)
   **AS-IS**
   ```
   -- The captured group can be referenced with $int
   SELECT REGEXP_REPLACE ('hello cubrid','([[:alnum:]]+)','$1!');
   ```
   **TO-BE**
      - Add backslashes to brackets
   ```
   -- The captured group can be referenced with $int
   SELECT REGEXP_REPLACE ('hello cubrid','\([[:alnum:]]+\)','$1!');
   ```

3. Example of "Groups" (269 lines)
   **AS-IS**
   ```
   -- ([a-c]{3}) creates a backreference, \1
   SELECT REGEXP_SUBSTR ('abckabcjabc', '([a-c]{3})k\1j\1');
   ```
   **TO-BE**
      - I do not know.

4. Example of "Assertions" (330 lines)
   **AS-IS**
   ```
   -- (?=subpattern): positive lookahead
   SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid(?= sql)', 'CUBRID');

   -- (?!subpattern): nagative lookahead
   SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid(?! sql)', 'CUBRID');
   ```
   **TO-BE**
      - Add backslashes to brackets
   ```
   -- (?=subpattern): positive lookahead
   SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid\(?= sql\)', 'CUBRID');

   -- (?!subpattern): nagative lookahead
   SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid\(?! sql\)', 'CUBRID');
   ```

5. Example of "POSIX-based character classes" (513 lines)
   **AS-IS**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_REPLACE ('가나다 가나 가나다라', '\b[[:alpha:]]{2}\b', '#');
       
   ::
   
       regexp_replace('가나다 가나 가나다라', '\b[[:alpha:]]{2}\b', '#')
       ======================
         '가나다 # 가나다라'
   ```
   **TO-BE**
      - Change Korean example to English
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_REPLACE ('abc ab abcd', '\b[[:alpha:]]{2}\b', '#');
       
   ::
   
       regexp_replace('abc ab abcd', '\b[[:alpha:]]{2}\b', '#')
       ======================
         'abc # abcd'
   ```

6. Example of "POSIX-based character classes" (513 lines)
   **AS-IS**
   ```
   .. code-block:: sql
   
       -- ;set regexp_engine=re2
       SELECT REGEXP_COUNT('가나 가나다라 마바사아 자차카타 파하', '\p{Hangul}+');
   
   ::
   
       regexp_count(_utf8'가나 가나다라 마바사아 자차카타 파하' collate utf8_ko_cs, _utf8'\p{Hangul}+' collate utf8_ko_cs)
       ==============================
       5
   ```
   **TO-BE**
      - Add backslashes to brackets
   ```
   .. code-block:: sql
   
       -- ;set regexp_engine=re2
       SELECT REGEXP_COUNT('ab abcd efgh ijkl mn', '[[:alpha:]]+');
   
   ::
   
       regexp_count('ab abcd efgh ijkl mn' collate utf8_ko_cs, '[[:alpha:]]+' collate utf8_ko_cs)
       ==============================
       5
   ```

7. Example of "REGEXP, RLIKE" (908 lines)
   **AS-IS**
   ```
   SELECT name FROM athlete where name REGEXP '^[a-d]';
   ```
   **TO-BE**
   ```
   SELECT name FROM public.athlete where name REGEXP '^[a-d]';
   ```

8. Example of "REGEXP, RLIKE" (908 lines)
   **AS-IS**
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_COUNT('가나123abc가다abc가가','[가-나]+');
       
   ::
   
       regexp_count('가나123abc가다abc가가','[가-나]+')
       ======================
         2
   ```
   **TO-BE**
      - result is different
   ```
   .. code-block:: sql
   
       SET NAMES utf8 COLLATE utf8_ko_cs;
       SELECT REGEXP_COUNT('가나123abc가다abc가가','[가-나]+');
       
   ::
   
       regexp_count('가나123abc가다abc가가','[가-나]+')
       ======================
         3
   ```

9. Example of "REGEXP_INSTR" (1064 lines)
   **AS-IS**
   ```
   regexp_instr('12354abc5','[:alpha:]+', 1, 1, 1);
   ======================
     9
   ```
   **TO-BE**
      - result is different
   ```
   regexp_instr('12354abc5','[:alpha:]+', 1, 1, 1);
   ======================
     7
   ```

youngjinj commented on en/admin/config.rst:
```diff
@@ -2017,6 +2019,8 @@ The following are other parameters. The type and value range for each parameter
 +-------------------------------------+--------+----------------+----------------+----------------+
 | supplemental_log                    | int    | 0 (off)        | 0              | 2              |
 +-------------------------------------+--------+----------------+----------------+----------------+
+| regexp_engine                       | string | RE2            |                |                |
```
It's trivial.

How about unifying it with lower case letters?
```suggestion
| regexp_engine                       | string | re2            |                |                |
```

youngjinj commented:
I hope to resolve the warning.
```
$ git diff ko/admin/config.rst
diff --git a/ko/admin/config.rst b/ko/admin/config.rst
index 4d4fc0b..10532d9 100644
--- a/ko/admin/config.rst
+++ b/ko/admin/config.rst
@@ -2055,9 +2055,9 @@ HA 관련 파라미터

 **java_stored_procedure_uds**

-    **java_stored_procedure_uds** 는 자바 저장 프로시저를 호출하는 cub_javasp 프로세스와 cub_server 프로세스 간에 TCP 연결 대신 유닉스 도메인 소켓을 통한 연결을 사용하기 위한 파라미터이다. 기본값은 **yes**이고 Windows의 경우 파라미터의 값과 관련 없이 **no**로 동작한다.
+    **java_stored_procedure_uds** 는 자바 저장 프로시저를 호출하는 cub_javasp 프로세스와 cub_server 프로세스 간에 TCP 연결 대신 유닉스 도메인 소켓을 통한 연결을 사용하기 위한 파라미터이다. 기본값은 **yes**\이고 Windows의 경우 파라미터의 값과 관련 없이 **no**\로 동작한다.

-        .. note::
+    .. note::

         cub_javasp 프로세스의 유닉스 도메인 소켓 파일 경로를 지정하는 **CUBRID_TMP** 환경 변수에 대한 내용은 :doc:`/env` 을 참고한다.

@@ -2186,7 +2186,7 @@ HA 관련 파라미터

 **regexp_engine**

- **regexp_engine** 는 정규식 연산자와 함수가 수행할 라이브러리를 선택하는 파라미터이다. **cppstd** 또는 **re2**를 설정할 수 있고 기본값은 **re2**이다. 정규 표현식 기능에 관한 자세한 내용은 :doc:`/sql/function/regex_fn` 을 참고한다.
+ **regexp_engine** 는 정규식 연산자와 함수가 수행할 라이브러리를 선택하는 파라미터이다. **cppstd** 또는 **re2**\를 설정할 수 있고 기본값은 **re2**\이다. 정규 표현식 기능에 관한 자세한 내용은 :doc:`/sql/function/regex_fn` 을 참고한다.

 .. _broker-configuration:
```

youngjinj commented on en/admin/config.rst:
```diff
@@ -379,6 +379,8 @@ On the below table, if "Applied" is "server parameter", that parameter affects t
 |                               +-------------------------------------+-------------------------+---------+----------+--------------------------------+-----------------------+
 |                               | supplemental_log                    | client/server parameter |         | int      | 0                              |                       |
+|                               +-------------------------------------+-------------------------+---------+----------+--------------------------------+-----------------------+
+|                               | regex_engine                        | client/server parameter |         | string   | re2                            | available             |
```
typo.
```suggestion
|                               | regexp_engine                        | client/server parameter |         | string   | re2                            | available             |
```

youngjinj commented on en/admin/config.rst:
```diff
@@ -2209,6 +2213,10 @@ The following are other parameters. The type and value range for each parameter
 
     **supplemental_log** is a parameter to determine whether information needed to support the CDC (Change Data Capture) or :ref:`flashback` is written to the log volume. CDC and flashback must be able to see how transactions logically changed the database through the physical logs. Any additional information required to interpret the physical logs is saved as **supplemental_log** . Setting this  ...
 
+**regexp_engine**
```
I think it should be mentioned that the types of values that can be set are "re2" and "cppstd".
No mention of "cppstd".

hgryoo replied:
The following paragraph is written. Wouldn't that be enough?
`**cppstd** or **re2** can be set and the default value is **re2**. `

youngjinj replied:
I didn't see the paragraph below. Enough.

hgryoo committed:
fix by code review

hgryoo committed:
some syntax in assertions only supported in <regex>

hgryoo committed:
fix differences between en/ko manual

hgryoo commented:
> I hope to resolve the warning.
> 
> ```
> $ git diff
> diff --git a/ko/sql/function/regex_fn.rst b/ko/sql/function/regex_fn.rst
> index 4515ac9..ba611bf 100644
> --- a/ko/sql/function/regex_fn.rst
> +++ b/ko/sql/function/regex_fn.rst
> @@ -15,7 +15,7 @@
>  .. _regex-syntax:
> 
>  정규 표현식 문법
> -=======
> +================
> 
>  정규 표현식을 구현하기 위해 CUBRID는 Google RE2 라이브러리와 C++ <regex> 표준 라이브러리를 사용한다.
>  **regexp_engine** 시스템 파라미터를 설정하여 정규 표현식 라이브러리를 선택할 수 있고 Google RE2 라이브러리를 기본으로 사용한다.
> @@ -75,18 +75,18 @@
>  +----------------+----------------------------------------------------------------------------------------------------------+
>  | \\r            | 캐리지 리턴 문자 (\\u000D 와 동일)                                                                       |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \\c\ *letter*  | 코드 단위 값이 *letter*의 코드 단위 값을                                                                 |
> +| \\c\ *letter*  | 코드 단위 값이 *letter*\의 코드 단위 값을                                                                |
>  |                | 32로 나눈 나머지와 같은 제어 코드 문자.                                                                  |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \\x\ *hh*      | 코드 단위 값이 2개의 16진수 *hh*에 해당하는 16진수 값을 갖는 문자                                        |
> +| \\x\ *hh*      | 코드 단위 값이 2개의 16진수 *hh*\에 해당하는 16진수 값을 갖는 문자                                       |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \\u\ *hhhh*    | 코드 단위 값이 4자리 16진수 *hhhh*에 해당하는 16진수 값을 갖는 문자 *C++ <regex>만 지원*                 |
> +| \\u\ *hhhh*    | 코드 단위 값이 4자리 16진수 *hhhh*\에 해당하는 16진수 값을 갖는 문자 *C++ <regex>만 지원*                |
>  +----------------+----------------------------------------------------------------------------------------------------------+
>  | \\x\ {*digits*}| *digits* 값에 해당하는 16진수 값을 갖는 문자 *Google RE2만 지원*                                         |
>  +----------------+----------------------------------------------------------------------------------------------------------+
>  | \\0            | 널 문자 (\\u0000와 동일). *C++ <regex>만 지원*                                                           |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \\\ *num*      | 여는 괄호가 *num*번째인 부분 일치의 결과. 자세한 내용은 아래 그룹을 참조.                                |
> +| \\\ *num*      | 여는 괄호가 *num*\번째인 부분 일치의 결과. 자세한 내용은 아래 그룹을 참조.                               |
>  +----------------+----------------------------------------------------------------------------------------------------------+
>  | \\d            | 10진수 문자 ([[:digit:]] 와 동일).                                                                       |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> @@ -104,11 +104,11 @@
>  | \\\ *character*| | 위의 특수 문자 시퀀스를 형성하는 문자를 제외한 모든 문자를 이스케이프 한다.                            |
>  |                | | 필요한 대상: ^ $ \\ . * + ? ( ) [ ] { } \|                                                             |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \[\ *class*\]  | *class*의 일부인 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                              |
> +| \[\ *class*\]  | *class*\의 일부인 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                             |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \[^\ *class*\] | *class*의 일부가 아닌 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                         |
> +| \[^\ *class*\] | *class*\의 일부가 아닌 문자열. 아래의 :ref:`regex-posix-character-class`\를 참조.                        |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> -| \\p{*class*}   | *class*의 유니코드 범위를 가지는 문자열. 아래의 :ref:`regex-unicode-character-class`\를 참조.            |
> +| \\p{*class*}   | *class*\의 유니코드 범위를 가지는 문자열. 아래의 :ref:`regex-unicode-character-class`\를 참조.           |
>  +----------------+----------------------------------------------------------------------------------------------------------+
> 
>  .. code-block:: sql
> @@ -164,11 +164,11 @@
>  +----------------+-------------------------------------------------------------------------+
>  | ?              | 선행하는 문자 또는 패턴이 선택사항 (0회 또는 1회 일치)                  |
>  +----------------+-------------------------------------------------------------------------+
> -| {*num*}        | 선행하는 문자 또는 패턴이 정확히 *num* 회 일치                          |
> +| {*num*}        | 선행하는 문자 또는 패턴이 정확히 *num*\회 일치                          |
>  +----------------+-------------------------------------------------------------------------+
> -| {*num,*}       | 선행하는 문자 또는 패턴이 정확히 *num* 회 또는 그 이상 일치             |
> +| {*num,*}       | 선행하는 문자 또는 패턴이 정확히 *num*\회 또는 그 이상 일치             |
>  +----------------+-------------------------------------------------------------------------+
> -| {*min,max*}    | 선행하는 문자 또는 패턴이 최소 *min* 회 일치하지만 *max*를 넘지 않음    |
> +| {*min,max*}    | 선행하는 문자 또는 패턴이 최소 *min*\회 일치하지만 *max*\를 넘지 않음   |
>  +----------------+-------------------------------------------------------------------------+
> 
>  .. code-block:: sql
> @@ -225,7 +225,7 @@
>        1
> 
>  기본적으로 이러한 모든 수량자는 가능한 한 조건을 충족하는 많은 문자를 사용하는 *greedy* 방식으로 수행된다.
> -그리고 이 동작은 한정자 뒤에 물음표(?)를 추가하여 *non-greedy*로 재정의할 수 있다.
> +그리고 이 동작은 한정자 뒤에 물음표(?)를 추가하여 *non-greedy*\로 재정의할 수 있다.
> 
>  .. code-block:: sql
> 
> @@ -371,7 +371,7 @@
>      ==============================
>        0
> 
> -정규식은 구분자 연산자(alternatvies, |)로 간단히 구분하여 여러 대체 패턴을 포함할 수 있다.
> +정규식은 구분자 연산자(alternatvies, \|)로 간단히 구분하여 여러 대체 패턴을 포함할 수 있다.
>  정규식은 대체 패턴이 일치하는 경우 일치하는 즉시 매칭한다.
>  하위 패턴(그룹 또는 어서션)은 구분자 연산자를 사용하여 여러 다른 패턴을 구분할 수도 있다.
> 
> @@ -538,7 +538,7 @@ POSIX 기반 문자 클래스(*[:classname:]*)는 아래와 같이 문자의 범
>  ~~~~~~~~~~~~~~~~~~~~~~~
> 
>  유니코드 문자 클래스(*\\p{classname}*)는 Google RE2 에서만 제공한다. 아래의 표와 같이 클래스 이름을 지정할 수 있다.
> -예를 들어 한글 문자를 매칭하기 위해서는 **\\p{Hangul}**을 사용한다.
> +예를 들어 한글 문자를 매칭하기 위해서는 **\\p{Hangul}**\을 사용한다.
> 
>  +-------------------------+
>  | 유니코드 문자 클래스    |
> @@ -886,8 +886,8 @@ POSIX 기반 문자 클래스(*[:classname:]*)는 아래와 같이 문자의 범
>  REGEXP, RLIKE
>  =============
> 
> -**REGEXP**, **RLIKE**\ 는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*이 *pattern*과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
> -*expression* 또는 *pattern*이 **NULL**이면 **NULL**을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**을 사용하는 세 번째 구문과 동일한 의미를 갖는다.
> +**REGEXP**, **RLIKE**\는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*\이 *pattern*\과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
> +*expression* 또는 *pattern*\이 **NULL**\이면 **NULL**\을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**\을 사용하는 세 번째 구문과 동일한 의미를 갖는다.
> 
>  ::
> 
> @@ -944,7 +944,7 @@ REGEXP_COUNT
> 
>  .. function:: REGEXP_COUNT (string, pattern_string [, position [, match_type]])
> 
> -    **REGEXP_COUNT** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*의 매칭 횟수를 반환한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> +    **REGEXP_COUNT** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\의 매칭 횟수를 반환한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> 
>      :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
>      :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
> @@ -1095,7 +1095,7 @@ REGEXP_LIKE
> 
>  .. function:: REGEXP_LIKE (string, pattern_string [, match_type])
> 
> -    **REGEXP_LIKE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*을 검색한다. 정규식 패턴이 *pattern_string*의 어느 곳에서나 일치하면 1이 반환된다. 그렇지 않으면 0이 반환된다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> +    **REGEXP_LIKE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\을 검색한다. 정규식 패턴이 *pattern_string*\의 어느 곳에서나 일치하면 1이 반환된다. 그렇지 않으면 0이 반환된다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> 
>      :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
>      :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
> @@ -1156,7 +1156,7 @@ REGEXP_REPLACE
> 
>  .. function:: REGEXP_REPLACE (string, pattern_string, replacement_string [, position [, occurrence [, match_type]]])
> 
> -    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*을 검색하여 문자열 *replacement_string*으로 바꾼다. **NULL**이 인수로 지정된 경우 **NULL**을 반환한다.
> +    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\을 검색하여 문자열 *replacement_string*\으로 바꾼다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> 
>      :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
>      :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
> @@ -1235,7 +1235,7 @@ REGEXP_SUBSTR
> 
>  .. function:: REGEXP_SUBSTR (string, pattern_string [, position [, occurrence [, match_type]]])
> 
> -    **REGEXP_SUBSTR** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*에 일치하는 문자열을 추출한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> +    **REGEXP_SUBSTR** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\에 일치하는 문자열을 추출한다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
> 
>      :param string: 입력 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다
>      :param pattern_string: 매칭을 수행할 정규식 패턴 문자열을 지정한다. 값이 **NULL**\이면 **NULL**\을 반환한다.
> ```

Thank you for the detailed reviews! I've added '\' all occurrences before a Korean character.

hgryoo commented:
> Sorry for not reviewing properly before.
> 
> The examples below do not work as expected. Below is for "en". "ko" needs to be checked.
> 
> 1. Example of "Special Pattern Characters" (140 lines)
>    **AS-IS**
>    ```
>    .. code-block:: sql
>    
>        -- \n : match a special character, when no_backslash_escapes=no
>        SELECT ('new\nline' REGEXP 'new
>        line');
>    
>    ::
>        
>        ('new
>        line' regexp 'new
>        line')
>        =====================================
>          1
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * Use new line character
>    
>    ```
>    .. code-block:: sql
>    
>        -- \n : match a special character, when no_backslash_escapes=no
>        SELECT ('new
>        line' REGEXP 'new
>        line');
>    
>    ::
>        
>        ('new
>        line' regexp 'new
>        line')
>        =====================================
>          1
>    ```
> 2. Example of "Groups" (255 lines)
>    **AS-IS**
>    ```
>    -- The captured group can be referenced with $int
>    SELECT REGEXP_REPLACE ('hello cubrid','([[:alnum:]]+)','$1!');
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * Add backslashes to brackets
>    
>    ```
>    -- The captured group can be referenced with $int
>    SELECT REGEXP_REPLACE ('hello cubrid','\([[:alnum:]]+\)','$1!');
>    ```
> 3. Example of "Groups" (269 lines)
>    **AS-IS**
>    ```
>    -- ([a-c]{3}) creates a backreference, \1
>    SELECT REGEXP_SUBSTR ('abckabcjabc', '([a-c]{3})k\1j\1');
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * I do not know.
> 4. Example of "Assertions" (330 lines)
>    **AS-IS**
>    ```
>    -- (?=subpattern): positive lookahead
>    SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid(?= sql)', 'CUBRID');
>    
>    -- (?!subpattern): nagative lookahead
>    SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid(?! sql)', 'CUBRID');
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * Add backslashes to brackets
>    
>    ```
>    -- (?=subpattern): positive lookahead
>    SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid\(?= sql\)', 'CUBRID');
>    
>    -- (?!subpattern): nagative lookahead
>    SELECT REGEXP_REPLACE ('cubrid dbms cubrid sql cubrid rdbms', 'cubrid\(?! sql\)', 'CUBRID');
>    ```
> 5. Example of "POSIX-based character classes" (513 lines)
>    **AS-IS**
>    ```
>    .. code-block:: sql
>    
>        SET NAMES utf8 COLLATE utf8_ko_cs;
>        SELECT REGEXP_REPLACE ('가나다 가나 가나다라', '\b[[:alpha:]]{2}\b', '#');
>        
>    ::
>    
>        regexp_replace('가나다 가나 가나다라', '\b[[:alpha:]]{2}\b', '#')
>        ======================
>          '가나다 # 가나다라'
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * Change Korean example to English
>    
>    ```
>    .. code-block:: sql
>    
>        SET NAMES utf8 COLLATE utf8_ko_cs;
>        SELECT REGEXP_REPLACE ('abc ab abcd', '\b[[:alpha:]]{2}\b', '#');
>        
>    ::
>    
>        regexp_replace('abc ab abcd', '\b[[:alpha:]]{2}\b', '#')
>        ======================
>          'abc # abcd'
>    ```
> 6. Example of "POSIX-based character classes" (513 lines)
>    **AS-IS**
>    ```
>    .. code-block:: sql
>    
>        -- ;set regexp_engine=re2
>        SELECT REGEXP_COUNT('가나 가나다라 마바사아 자차카타 파하', '\p{Hangul}+');
>    
>    ::
>    
>        regexp_count(_utf8'가나 가나다라 마바사아 자차카타 파하' collate utf8_ko_cs, _utf8'\p{Hangul}+' collate utf8_ko_cs)
>        ==============================
>        5
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * Add backslashes to brackets
>    
>    ```
>    .. code-block:: sql
>    
>        -- ;set regexp_engine=re2
>        SELECT REGEXP_COUNT('ab abcd efgh ijkl mn', '[[:alpha:]]+');
>    
>    ::
>    
>        regexp_count('ab abcd efgh ijkl mn' collate utf8_ko_cs, '[[:alpha:]]+' collate utf8_ko_cs)
>        ==============================
>        5
>    ```
> 7. Example of "REGEXP, RLIKE" (908 lines)
>    **AS-IS**
>    ```
>    SELECT name FROM athlete where name REGEXP '^[a-d]';
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    ```
>    SELECT name FROM public.athlete where name REGEXP '^[a-d]';
>    ```
> 8. Example of "REGEXP, RLIKE" (908 lines)
>    **AS-IS**
>    ```
>    .. code-block:: sql
>    
>        SET NAMES utf8 COLLATE utf8_ko_cs;
>        SELECT REGEXP_COUNT('가나123abc가다abc가가','[가-나]+');
>        
>    ::
>    
>        regexp_count('가나123abc가다abc가가','[가-나]+')
>        ======================
>          2
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * result is different
>    
>    ```
>    .. code-block:: sql
>    
>        SET NAMES utf8 COLLATE utf8_ko_cs;
>        SELECT REGEXP_COUNT('가나123abc가다abc가가','[가-나]+');
>        
>    ::
>    
>        regexp_count('가나123abc가다abc가가','[가-나]+')
>        ======================
>          3
>    ```
> 9. Example of "REGEXP_INSTR" (1064 lines)
>    **AS-IS**
>    ```
>    regexp_instr('12354abc5','[:alpha:]+', 1, 1, 1);
>    ======================
>      9
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    **TO-BE**
>    
>    * result is different
>    
>    ```
>    regexp_instr('12354abc5','[:alpha:]+', 1, 1, 1);
>    ======================
>      7
>    ```

Fixed. Some syntaxes in assertions and a backreference of groups are not supported in Google RE2. I've added `note` or `warning` blocks

hgryoo committed:
remove old description only stayed in korean manual

ssihil commented on en/sql/function/regex_fn.rst:
```diff
@@ -240,7 +258,7 @@ Groups allow to apply quantifiers to a sequence of characters (instead of a sing
     -- The captured group can be referenced with $int
-    SELECT REGEXP_REPLACE ('hello cubrid','([[:alnum:]]+)','$1!');
+    SELECT REGEXP_REPLACE ('hello cubrid','\([[:alnum:]]+\)','$1!');
 
```
The Korean manual also needs revision.

youngjinj commented:
I hope to resolve the warning.
- *non-greedy*로 → *non-greedy*\로
```
Check warning on line 227 in ko/sql/function/regex_fn.rst

GitHub Actions
/ build_docs (ko)

Inline emphasis start-string without end-string.
```

youngjinj commented on en/sql/function/regex_fn.rst:
```diff
@@ -473,11 +508,17 @@ The POSIX-based character class (*[:classname:]*) defines categories of characte
 +------------+-----------------------------------------+
-| [:d:]      | Decimal digit character                 |
+
+[:d:], [:w:] and [:s:] are an extension to the ECMAScript grammar. C++ <regex> 에서만 사용할 수 있다.
```
Korean should not appear in English manuals.

ssihil commented on en/sql/function/regex_fn.rst:
```diff
@@ -486,21 +527,371 @@ The POSIX-based character class (*[:classname:]*) defines categories of characte
 
-    regexp_substr('Samseong-ro 86-gil, Gangnam-gu, Seoul 06178', '[[:digit:]]{5}')
+    regexp_substr('Samseong-ro 86-gil, Gangnam-gu, Seoul 06178', '[[:digit:]]\{5\}')
     ================================
```
Backslashes are not included in the sql execution result, so they need to be deleted.
regexp_substr('Samseong-ro 86-gil, Gangnam-gu, Seoul 06178', '[[:digit:]]{5}')

ssihil commented on en/sql/function/regex_fn.rst:
```diff
@@ -486,21 +527,371 @@ The POSIX-based character class (*[:classname:]*) defines categories of characte
 ::
 
-    regexp_replace('가나다 가나 가나다라', '\b[[:alpha:]]{2}\b', '#')
+    regexp_replace('가나다 가나 가나다라', '\b[[:alpha:]]\{2\}\b', '#')
```
The result is as below.

  regexp_replace('가나다 가나 가나다라' collate utf8_ko_cs, '\b[[:alpha:]]{2}\b' collate utf8_ko_cs, '#' collate utf8
_ko_cs)

======================
  '가나다 가나 가나다라'

hgryoo replied:
`;set regexp_engine=cppstd` is required. It is described at the beginning of the section.

ssihil replied:
Checked.
I think you need to add "collate..." to the result header.

ssihil replied:
After "sSET NAMES utf8 COLLATE utf8_ko_cs;", the result header must contain "Collate ...".

hgryoo replied:
Fixed, to apply [[:alpha:]] properly, I've added braces `(`, `)`.

ssihil commented on ko/sql/function/regex_fn.rst:
```diff
@@ -501,13 +543,362 @@ POSIX 기반 문자 클래스(*[:classname:]*)는 아래와 같이 문자의 범
-**REGEXP**, **RLIKE**\ 는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*이 *pattern*과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
-*expression* 또는 *pattern*이 **NULL**이면 **NULL**을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**을 사용하는 세 번째 구문과 동일한 의미를 갖는다.
+**REGEXP**, **RLIKE**\ 는 동일한 의미로 사용되며, 정규 표현식을 이용한 패턴을 매칭한다. 아래의 문법에서, *expression*\이 *pattern*\과 일치하면 1을 반환한다. 그렇지 않으면 0을 반환한다.
+*expression* 또는 *pattern*이 **NULL**\이면 **NULL**\을 반환한다. 두 번째 구문은 두 구문 모두 **NOT**\을 사용하는 세 번째 구문과 동일한 의미를 갖는다.
```
Add backslashe
*pattern*\이

hgryoo replied:
Fixed. thank you

ssihil commented on ko/sql/function/regex_fn.rst:
```diff
@@ -775,11 +1165,11 @@ REGEXP_REPLACE
 .. function:: REGEXP_REPLACE (string, pattern_string, replacement_string [, position [, occurrence [, match_type]]])
 
-    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*을 검색하여 문자열 *replacement_string*으로 바꾼다. **NULL**이 인수로 지정된 경우 **NULL**을 반환한다.
+    **REGEXP_REPLACE** 함수는 주어진 문자열 *string* 내에서 정규식 패턴 *pattern_string*\을 검색하여 문자열 *replacement_string*으로 바꾼다. **NULL**\이 인수로 지정된 경우 **NULL**\을 반환한다.
```
Add backslashe
*replacement_string*\으로

hgryoo replied:
fixed

hgryoo committed:
slip to translate in english manual

hgryoo committed:
fix result

hgryoo committed:
add escape before korean

hgryoo committed:
remove escape in example

hgryoo committed:
add brace, add collate.. in the result of example

hgryoo committed:
add collate.. in the example results of each function
